how-to/rotate/github-apps.rst

```diff
@@ -0,0 +1,22 @@
+Rotate Github App Private Keys
+==============================
+
+Releng uses Github Apps for authentication in a variety of places. This page
+outlines how to rotate the private keys associated with a Github App that are
+used to `generate an app installation token`_.
+
+Steps to Rotate
+---------------
+
+1. Open https://github.com/organizations/mozilla-releng/settings/apps and click
+   ``Edit`` next to the app you are rotating.
+2. Scroll down to the ``Private Keys`` section and click the ``Generate a
+   private key`` button. This will prompt a download of the private key and
+   associate the public key with the app.
+3. Press ``Delete`` on the old key(s) you are rotating.
```

Comment:
What are the implications of deleting this key? Do all of the installation tokens need to be regenerated after this? (If so, is there a way to do this gracefully?)

Reply:
Installation tokens expire after 1 hour, so yes deleting the key would means use cases will stop working.

Good call out though, I'll insert a step about updating consumers before pressing delete.

Reply:
I just swapped step 3 and 4 around.

Reply:
Should we be waiting 1h before deleting, or is there no risk to deleting immediately?

Reply:
I believe access tokens created from the deleted key would still be valid, as the key is only used to sign a JWT, which is then used to generate the access token. From that point on, the access token is used to authenticate and the JWT is no longer used. So I don't believe it would be re-checked on subsequent API access.

I guess it's possible that Github internally maps JWTs <-> access tokens, and re-checks that the JWT signature is still valid.. but that feels unlikely, and I can't find anything that would suggest the access token would get invalidated.

All that to say.. I believe there's no risk to deleting immediately, but I'm not 100% sure.

Reply:
OK, this seems totally fine for now then...we can adjust later if needed.